### ISSUE TITLE
client/ASYNC_VERIFY::verify_chunk improvements

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#1197]
**Reported by rctay on 18 Mar 43120222 11:49 UTC**
The attached patches are git-am-friendly and were made on top of boinc trunk.

Migrated-From: http://boinc.berkeley.edu/trac/ticket/1225

**Comment by romw** [comment]
**Commented by rctay on 2 Jul 43120228 16:26 UTC**

```
[1/2](PATCH) client/ASYNC_VERIFY::verify_chunk: delay casting of size_t to int

We only need ints when calling md5_append(), so do the casting at
call-time. This reduces the number of casts needed, as well as silences
a compiler warning on VS 2005.

We need not worry about integer overflows, as md5_append checks if the
int argument is negative.
```

```
[2/2](PATCH) client/ASYNC_VERIFY::verify_chunk: check ferror() after a fread()/fwrite()
```

**Comment by ChristianBeer** [comment]
Original patches from 4 years ago attached. May need to be adjusted for current master.

[0001-client-ASYNC_VERIFY-verify_chunk-delay-casting-of-si.patch.txt](https://github.com/BOINC/boinc/files/910616/0001-client-ASYNC_VERIFY-verify_chunk-delay-casting-of-si.patch.txt)
[0002-client-ASYNC_VERIFY-verify_chunk-check-ferror-after-.patch.txt](https://github.com/BOINC/boinc/files/910615/0002-client-ASYNC_VERIFY-verify_chunk-check-ferror-after-.patch.txt)

**Comment by davidpanderson** [comment]
Not needed.

**Comment by ChristianBeer** [comment]
I disagree. The changes will make the logic more robust and also cut down on compiler warnings.

**Comment by adamradocz** [comment]
The patch has been merged in #2697, this issue can be closed.
